Asgard/Communications/Classes/CbusMessenger.cs

```diff
@@ -64,6 +69,7 @@ private void HandleTransportMessage(object? sender, MessageReceivedEventArgs e)
             try
             {
                 var frame = this.cbusCanFrameProcessor.ParseFrame(e.Message);
+                if (frame?.Message is null) return;
```

Comment:
On this one - I wonder if we'd be better off going and refactoring the `CbusCanFrame` itself at some point to make it so that the `Message` property can never be null - could have `ParseFrame` return an interface of some sort that makes that guarantee rather than returning the actual frame object etc

Reply:
Yes, that seems a sensible approach. I'll take a look.

Reply:
Certainly something to keep an eye on as we go through things, rather than marking things as nullable

Reply:
Looked more closely. Message needs to be nullable because it isn't set in the CbusCanFrame constructor. It's only set from outside the class, so I can't see how it can be not nullable.

Reply:
We have an interface already actually - if `ParseFrame` is updated to return `ICbusCanFrame`, and that interface itself is updated to have a non-nullable `Message` get-only property, it seems to like that

Reply:
In reality neither frame or frame.Message will be null at that point. The test is simply to keep the compiler happy. It probably needs a comment.

Reply:
oh, except just seen a different warning pop up... never mind :D

Reply:
yeah - just don't like adding code just to keep compilers happy, usually a sign that things can be improved elsewhere.  It's something to come back to at some point

Reply:
Perhaps throwing an exception would be a better outcome than just returning?

Reply:
Yep - I've got a few TODOs in some of the stuff I've added recently that mention the possibility of throwing since it's a scenario that just shouldn't happen